controllers/provisioning_controller.go

```diff
@@ -477,37 +481,44 @@ func (r *ProvisioningReconciler) networkStackFromServiceNetwork(ctx context.Cont
 	return ns, nil
 }
 
-func getHostByProviderId(provId string) string {
-	if provId == "" {
+func getHostByMachine(meta metav1.ObjectMeta) string {
+	annotations := meta.GetAnnotations()
+	annotation, ok := annotations[HostAnnotation]
```

Comment:
The reason I did it this way is because I did not see `ProviderID` on the machines from the customer's must-gather. I assume it's set later, just as the Node's ProviderID

Reply:
It should get set fairly promptly: https://github.com/openshift/cluster-api-provider-baremetal/blob/master/pkg/cloud/baremetal/actuators/machine/actuator.go#L235-L239

Reply:
Yeah, but `Update` is not called if `Exists` fails, and it fails here: https://github.com/openshift/cluster-api-provider-baremetal/blob/master/pkg/cloud/baremetal/actuators/machine/actuator.go#L316 (because the state is None, because BMO cannot start, because we need MACs).

Reply:
Yep, just looked at the logs and saw the same thing.
Arguably we could/should make an exception there for Hosts that have ExternallyProvisioned in the Spec and no provisioning state. I think that would mean that masters would never be prevented from getting nodes linked to machines because metal³ is not running.

Reply:
e.g. https://github.com/openshift/cluster-api-provider-baremetal/pull/183

Reply:
In fact we still depend on BMO to link machines and hosts, because it is needed to copy the HardwareDetails from the status annotation into the Status, whence the Machine can determine its IP addresses.